### DISCUSSION TITLE
Disable all instances of useQuery after initial fetch?

### DISCUSSION THREAD

**Discussion by Purus** [TanStack/query#2571] [Q&A]
Currently I am trying to disable re-fetching all queries globally. Individual queries will be configured if required..

I am using the below default options for the same. But I see that some status(stale, fetching) updates in the react-query developer toolbar.

More background: I am currently having a material-ui dialog box shown. When the stale status changed, the UI gets refreshed and so the dialog box also closes automatically.

Is any other config changes required to completely disable re-fetching at global level?

```typescript
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnMount: false,
        refetchOnWindowFocus: false,
        refetchInterval: false,
        refetchOnReconnect: false,
        cacheTime: Infinity,
        staleTime: Infinity
      }
    }
  });
```

This is similar to https://github.com/tannerlinsley/react-query/discussions/2270, but at a global level.

**Comment by TkDodo** [comment]
are you saying that a re-render of a component because the e.g. a status like `stale` is changing closes your dialog to close? I would say that this is a problem of the dialog then, not react-query :) a component should be able to re-render as often as it wants without any significant impact like that.

you can look into `notifyOnChangeProps: 'tracked'`, which will not re-render when these properties change unless you actively use them in your component, but it will likely only shift the problem

**Comment by Purus** [comment]
Great :-)

Using `notifyOnChangeProps: 'tracked'` solved my issue actually. I will monitor if this is going to impact any other functionality.. Thanks a ton.
